ui/src/services/loki/api.js

```diff
@@ -22,44 +21,39 @@ type StreamValue = {
 type LokiQueryResult = {
   status: 'success',
   data: {
-    resultType: 'matrix' | 'streams',
-    result: MetrixValue | StreamValue,
+    resultType: 'streams',
+    result: StreamValue,
     // Those statistics allow users to understand the amount of data processed and at which speed.
     // https://grafana.com/docs/loki/latest/api/#Statistics
     stats: {},
   },
 };
 
 // by default to get the last 7day alerts from Loki
-export function getLast7DaysAlerts(): LokiQueryResult {
-  if (!lokiApiClient) {
-    throw new Error('lokiApiClient should be defined');
-  }
-
-  const query = "{container='alertmanager-logger'}";
+export function getLast7DaysAlerts(): Promise<Alert[]> {
   const start = new Date(
     new Date().getTime() - 7 * 24 * 60 * 60 * 1000,
   ).toISOString();
   const end = new Date().toISOString();
 
-  return lokiApiClient
-    .get(`/api/v1/query_range?start=${start}&end=${end}&query=${query}`)
-    .then((resolve) => {
-      if (resolve.error) {
-        throw resolve.error;
-      }
-      return resolve;
-    });
+  return getAlertsLoki(start, end).then((result) =>
+    removeWarningAlerts(formatHistoryAlerts(result.data.result)),
```

Comment:
Maybe this can be done in getAlertsLoki in order to avoid having to do it in calling methods.